### DISCUSSION TITLE
SSR (React, Nuxt)

### DISCUSSION THREAD

**Discussion by joaocarvalhowd** [frontendbr/forum#748] [Perguntas e Respostas]
@mtmr0x Cara, acho que você pode me ajudar com essa dúvida.
Na evino, como você fez o SSR, para montar a página vocês fazem requisições na API usando async data ou foi de outra forma?

**Comment by felipefialho** [comment]
Porque fechou @joaocarvalhowd ?

**Comment by joaocarvalhowd** [comment]
@LFeh Achei uma outra issue falando um pouco sobre isso (#631).

**Comment by joaocarvalhowd** [comment]
Mas acho interessante saber como funcionou na evino essa questão, se puder contar um pouco @mtmr0x , ajudaria bastante tbm. Pois estou iniciando um novo projeto na empresa que estou, para transformar o e-commerce em SSR.

**Comment by mtmr0x** [comment]
@joaocarvalhowd Só para entender melhor, tu quer saber como lidei com fetch de dados para ter o conteúdo na página, certo?

Se sim, eu usei o Redux no server-side, numa abstração onde usei o ciclo de vida do componente para gerenciar o request com o mesmo código. Nesse artigo meu eu explico como foi feito esse gerenciamento com links de referência: https://medium.com/@mtmr0x/https-medium-com-matmarsiglio-the-worst-problems-youll-find-building-an-isomorphic-application-pt-2-5903baaa0e92

Espero que ajude! :)

**Comment by joaocarvalhowd** [comment]
@mtmr0x Era isso mesmo que eu gostaria de saber, aqui nós optamos por trabalhar com o Nuxt.JS, mas tinhamos dúvidas se o tratamento que ele fazia para isso, era a melhor forma. E pelo que você falou no medium e basicamente o mesmo tratamento que ele faz, certo?

Valeu pela ajuda!

**Comment by mtmr0x** [comment]
@joaocarvalhowd Usando uma biblioteca que pode fazer fetch no  server side, tu conseguirá fazer isso sem problema, mas o segredo está em como passar esses dados para o componente. No React, por exemplo, se for renderizado do servidor, preciso disparar a action no ciclo antes de o componente ser renderizado, já no cliente, é ao contrário, isso precisa ser feito após o render do componente ser executado.

Para cada biblioteca de virtual DOM que tu achar, tu vai ver particularidades diferentes. Se tu `debuggar` o ciclo do seu fetch de dados e acompanhar os status dos seus componentes no processo, tu vai conseguir mapear onde cada coisa deve ser chamada. 

E sim, é um trabalho bem clínico e analítico mesmo, sem resposta certa, pois depende do que está sendo usado no projeto e os efeitos colaterais dessas dependências.
